tests/test_venue_request_v2.py

```diff
@@ -550,7 +550,19 @@ def test_venue_recruitment(self, client, test_client, selenium, request_page, ve
             readers=['{}/Program_Chairs'.format(venue['venue_id']), venue['support_group_id']],
             signatures=['~SomeFirstName_User1'],
             writers=[]
-        ))
+        )
+
+        with pytest.raises(openreview.OpenReviewException, match=r'Invalid token: {{serve}} does not exist.'):
+           recruitment_note=test_client.post_note(recruitment_note)
+
+        recruitment_note.content['invitation_email_subject'] = '[' + venue['request_form_note'].content['Abbreviated Venue Name'] + '] Invitation to serve as {invitee_role'
+
+        with pytest.raises(openreview.OpenReviewException, match=r'Invalid token: {invitee_role. Tokens must be wrapped in double curly braces.'):
+           recruitment_note=test_client.post_note(recruitment_note)
+                                                               
+        recruitment_note.content['invitation_email_subject'] = '[' + venue['request_form_note'].content['Abbreviated Venue Name'] + '] Invitation to serve as {{invitee_role}}'
+        recruitment_note = test_client.post_note(recruitment_note)
```

Comment:
could you add a test where `{{contact_info}}` is in the subject? I believe the validation is passing but how is the email subject being formatted? is contact_info being replaced or not?

Reply:
Currently this function doesn't check for where these tokens are being used, it just checks the braces. From what I see in the code about how the replacements are being done, you're right, I don't think `{{contact_info}}` would get replaced in the subject. I can add this validation.

Reply:
I recommend having a dictionary to represent which tokens are being replaced for each field name and then to the validation based on that:

```
{
"invitation_email_subject": ["invitee_role"],
"invitation_email_content": ['fullname', 'invitee_role', 'invitation_url', 'contact_info', 'reviewer_name']
"accepted_email_template": ....
}
```

'accept_url', 'decline_url' are being deprecated so I wouldn't validate those.